src/Castle.Core/DynamicProxy/IProxyGenerator.cs

```diff
@@ -1029,5 +1029,70 @@ TClass CreateClassProxy<TClass>(ProxyGenerationOptions options, params IIntercep
 		object CreateClassProxy(Type classToProxy, Type[] additionalInterfacesToProxy,
 		                        ProxyGenerationOptions options,
 		                        object[] constructorArguments, params IInterceptor[] interceptors);
+
+		/// <summary>
+		///   Creates a delegate that, when invoked, calls the specified <paramref name="interceptors"/>.
+		/// </summary>
+		/// <param name="delegateToProxy">Type of delegate which will be proxied.</param>
+		/// <param name="interceptors">The interceptors called during the invocation of the delegate.</param>
+		/// <returns>
+		///   New delegate of type <paramref name="delegateToProxy"/> that, when invoked, calls the specified <paramref name="interceptors"/>.
+		/// </returns>
+		/// <exception cref="ArgumentNullException">Thrown when the given <paramref name="delegateToProxy"/> object is a null reference (Nothing in Visual Basic).</exception>
```

Comment:
I forget to mention "[...] or when `interceptors` is a null reference [...]". Same in other XML documentation comments. Going to rectify this soon.